tests/e2e/runner/kvtool.go

```diff
@@ -0,0 +1,104 @@
+package runner
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+)
+
+type KvtoolRunnerConfig struct {
+	KavaConfigTemplate string
+
+	ImageTag   string
+	IncludeIBC bool
+
+	EnableAutomatedUpgrade  bool
+	KavaUpgradeName         string
+	KavaUpgradeHeight       int64
+	KavaUpgradeBaseImageTag string
+
+	SkipShutdown bool
+}
+
+// KvtoolRunner implements a NodeRunner that spins up local chains with kvtool.
+// It has support for the following:
+// - running a Kava node
+// - optionally, running an IBC node with a channel opened to the Kava node
+// - optionally, start the Kava node on one version and upgrade to another
+type KvtoolRunner struct {
+	config KvtoolRunnerConfig
+}
+
+var _ NodeRunner = &KvtoolRunner{}
+
+func NewKvtoolRunner(config KvtoolRunnerConfig) *KvtoolRunner {
+	return &KvtoolRunner{
+		config: config,
+	}
+}
+
+func (k *KvtoolRunner) StartChains() Chains {
+	// install kvtool if not already installed
+	installKvtoolCmd := exec.Command("./scripts/install-kvtool.sh")
+	installKvtoolCmd.Stdout = os.Stdout
```

Comment:
I had to look it up -- forgot that stdout is not buffered in golang, so panic on error below won't hide logs 👍